arangod/RestHandler/RestBatchHandler.cpp

```diff
@@ -38,10 +39,103 @@
 #include "Scheduler/SchedulerFeature.h"
 #include "Utils/ExecContext.h"
 
+#include <absl/strings/str_cat.h>
+
 using namespace arangodb;
 using namespace arangodb::basics;
 using namespace arangodb::rest;
 
+bool arangodb::MultipartMessage::skipBoundaryStart(size_t& offset) const {
+  size_t foundPos = message.find(boundary, offset);
+  if (foundPos == offset) {
+    foundPos += boundary.size();
+    while (foundPos < message.size() && message[foundPos] == ' ') {
+      ++foundPos;
+    }
+    if (foundPos < message.size() && message[foundPos] == '\r') {
+      ++foundPos;
+    }
+    if (foundPos < message.size() && message[foundPos] == '\n') {
+      offset = foundPos + 1;
+      return true;
+    }
+  }
+  return false;
+}
+
+bool arangodb::MultipartMessage::findBoundaryEnd(size_t& offset) const {
+  size_t foundPos = message.find(boundary, offset);
+  if (foundPos != std::string_view::npos) {
+    offset = foundPos + boundary.size();
+    return true;
+  }
+  return false;
+}
+
+bool arangodb::MultipartMessage::isAtEnd(size_t& offset) const {
+  if (offset < message.size() && message[offset] == '-') {
+    ++offset;
+    if (offset < message.size() && message[offset] == '-') {
+      ++offset;
+      return true;
+    }
+  }
+  return false;
+}
```

Comment:
due to jetlag I'm not sure if it is a problem that `offset` is incremented by 1 if the next character is a `-`, but the next one is not. The old code checked this via `*p == '-' && *(p + 1) == '-'`.

Reply:
Good catch. Fixed it.